lib/Conversion/TorchToLinalg/Utils.cpp

```diff
@@ -67,6 +69,42 @@ Value torch_to_linalg::getZeroPaddedTensor(
   return getPaddedTensor(op, b, input, paddingInts, paddingInts, c0);
 }
 
+// Helper function that adds dynamic padding to a tensor, ignoring unpaddedDims
+// dimensions at the beginning. The high and low padding are the same, and the
+// padding value is zero.
+Value torch_to_linalg::getDynamicZeroPaddedTensor(
+    Operation *op, OpBuilder &b, Value &input, SmallVectorImpl<Value> &padding,
+    int unpaddedDims) {
+  assert(input.getType().isa<RankedTensorType>() &&
+         "input must be RankedTensorType");
+  Location loc = op->getLoc();
+
+  Value c1 = b.create<arith::ConstantOp>(loc, b.getIndexAttr(1));
+  SmallVector<Value> inputDims = getTensorSizes(b, loc, input);
+
+  Value c0 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(0));
+  SmallVector<Value> paddingIncludingUnchanged{c0, c0};
+  paddingIncludingUnchanged.append(padding);
+  for (auto pad = paddingIncludingUnchanged.begin();
+       pad < paddingIncludingUnchanged.end(); pad++)
+    *pad = castIntToIndex(b, loc, *pad);
+
+  auto inputRank = input.getType().cast<RankedTensorType>().getRank();
+  Type elementType = input.getType().cast<RankedTensorType>().getElementType();
+  Type inputType = RankedTensorType::get(
+      llvm::ArrayRef<int64_t>(SmallVector<int64_t>(inputRank, kUnknownSize)),
+      elementType);
+  input = b.create<tensor::CastOp>(loc, inputType, input);
+
+  SmallVector<Value> strides(inputRank, c1);
+  Value cf0 =
+      b.create<arith::ConstantOp>(loc, b.getFloatAttr(elementType, 0.0));
+  SmallVector<OpFoldResult> paddingValues =
+      getAsOpFoldResult(paddingIncludingUnchanged);
+  return tensor::createPadScalarOp(inputType, input, cf0, paddingValues,
+                                   paddingValues, false, loc, b);
```

Comment:
nit: add comments for arguments so that it's easier to follow. refer to https://github.com/llvm/torch-mlir/blob/9208bf0eb63e3589b54b23324b5202916436a220/lib/Conversion/TorchToLinalg/Utils.cpp#L49-L51